lib/screens/login/login_form.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:survey_flutter/screens/widgets/primary_button_style.dart';
+import 'package:survey_flutter/screens/widgets/primary_text_field_decoration.dart';
+import 'package:survey_flutter/theme/app_constants.dart';
+import 'package:survey_flutter/utils/keyboard_manager.dart';
+
+class LoginForm extends StatefulWidget {
+  const LoginForm({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _LoginFormState();
+}
+
+class _LoginFormState extends State<LoginForm> {
+  final _formKey = GlobalKey<FormState>();
+
+  TextTheme get _textTheme => Theme.of(context).textTheme;
+  AppLocalizations get _localizations => AppLocalizations.of(context)!;
+
+  TextFormField get _emailTextField => TextFormField(
+        keyboardType: TextInputType.emailAddress,
+        autocorrect: false,
+        decoration: PrimaryTextFieldDecoration(
+          hintText: _localizations.emailInputHint,
+          hintTextStyle: _textTheme.bodyMedium,
+        ),
+        style: _textTheme.bodyMedium,
+      );
+
+  TextFormField get _passwordTextField => TextFormField(
+        autocorrect: false,
+        obscureText: true,
+        decoration: PrimaryTextFieldDecoration(
+          hintText: _localizations.passwordInputHint,
+          hintTextStyle: _textTheme.bodyMedium,
+        ),
+        style: _textTheme.bodyMedium,
+      );
+
+  ElevatedButton get _loginButton => ElevatedButton(
+        style: PrimaryButtonStyle(hintTextStyle: _textTheme.labelMedium),
+        onPressed: _submit,
+        child: Text(_localizations.loginButton),
+      );
+
+  void _submit() {
+    KeyboardManager.dismiss(context);
+    // TODO: Integrate with API
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const fieldSpacing = 20.0;
```

Comment:
Should we put it in `metric`? 🤔

Reply:
@doannimble, As I see it, the `fieldSpacing` here is the vertical spacing between the `LoginForm` fields. It is very unlikely that other widgets use the same vertical spacing between their components. That's why it's internally scoped in this method.

It's different than the other `Metrics`'s constants, like `defaultHorizontalPadding` will be used for every screen; `defaultBorderRadius` might be used for primitive components like text field, text view, button, etc. They're a kind of design system definition which will be used across the app.

What do you think? 🙏

Reply:
@Thieurom Okay, in this case, we can define it as a private constant for this widget. Putting in the build function seem not to be a best way imo 🤔